server/lib/multipartRequest.js

```diff
@@ -0,0 +1,52 @@
+import Promise from 'bluebird';
+import { ArgumentError } from 'auth0-extension-tools';
+
+export default function(client, entity, opts = {}, perPage = 100, concurrency = 5) {
+  if (client === null || client === undefined) {
+    throw new ArgumentError('Must provide a auth0 client object.');
+  }
+
+  if (!entity && !client[entity]) {
+    throw new ArgumentError('Must provide a valid entity for auth0 client.');
+  }
+
+  const getter = client[entity].getAll;
+  const options = { ...opts, per_page: perPage };
+  const result = [];
+  let total = 0;
+  let pageCount = 0;
+
+  const getTotals = () =>
+    getter({ ...options, include_totals: true, page: 0 })
+      .then((response) => {
+        total = response.total || 0;
+        pageCount = Math.ceil(total / perPage);
+        const data = response[entity] || response || [];
```

Comment:
If `response[entity]` is undefined, why is the full `response` appropriate?

Reply:
Response can be either 
```
{
  total: 10,
  ...,
  connections: [ { connection }, ... ]
}
```
or 
```
[ { connection }, ... ]
```